app/bundles/CoreBundle/EventListener/UpgradeSubscriber.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * @package     Mautic
+ * @copyright   2016 Mautic Contributors. All rights reserved.
+ * @author      Mautic
+ * @link        http://mautic.org
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\CoreBundle\EventListener;
+
+use Mautic\CoreBundle\CoreEvents;
+use Mautic\CoreBundle\Event\UpgradeEvent;
+use Mautic\CoreBundle\Helper\UTF8Helper;
+use Mautic\CoreBundle\IpLookup\AbstractLookup;
+
+/**
+ * Class UpgradeSubscriber
+ */
+class UpgradeSubscriber extends CommonSubscriber
+{
+    /**
+     * @var AbstractLookup
+     */
+    private $ipLookupService;
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return array(
+            CoreEvents::POST_UPGRADE => array('postUpgrade', 0)
+        );
+    }
+
+    /**
+     * @param UpgradeEvent $event
+     *
+     * @return void
+     */
+    public function postUpgrade(UpgradeEvent $event)
+    {
+        $version = $this->factory->getKernel()->getVersion();
+
+        // In the upgrade to 1.3, we included a bugfix for corrupt ip data
+        if ($event->isSuccessful() && version_compare($version, '1.3', '<=')) {
+            $this->updateIpDetails();
+        }
+    }
+
+    /**
+     * Update the ip_details column to fix the bug in GH #1375
+     *
+     * @return void
+     */
+    private function updateIpDetails()
+    {
+        $connection = $this->factory->getEntityManager()->getConnection();
+        $query = $connection->createQueryBuilder()
+            ->select('*')
+            ->from(MAUTIC_TABLE_PREFIX . 'ip_addresses', 'ip');
+
+        $results = (array) $query->execute()->fetchAll();
```

Comment:
I'm afraid this can lead to hit the time or memory limit when you consider some people track websites with millions of visitors per month with Mautic.